src/textual/widgets/_list_view.py

```diff
@@ -286,13 +287,30 @@ def pop(self, index: Optional[int] = None) -> AwaitRemove:
             An awaitable that yields control to the event loop until
                 the DOM has been updated to reflect item being removed.
         """
-        if index is None:
-            await_remove = self.query("ListItem").last().remove()
-        else:
-            await_remove = self.query("ListItem")[index].remove()
-        return await_remove
-
-    def remove_items(self, indices: Iterable[int]) -> AwaitRemove:
+        if len(self) == 0:
+            raise IndexError("pop from empty list")
+
+        index = index if index is not None else -1
+        item_to_remove = self.query("ListItem")[index]
+        normalized_index = index if index >= 0 else index + len(self)
+
+        async def do_pop():
```

Comment:
Docstring and typing  por favor. Since its internal, and essentially private, the docstrings can be written for future Textual core-devs.

Reply:
I've added docstrings in 9314cde, but feel free to tweak!